src/main/java/core/basesyntax/AbstractFigure.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public abstract class AbstractFigure implements Figure {
+    private String color;
+
+    public AbstractFigure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
```

Comment:
Consider whether the `setColor` method is necessary. Figures are often immutable, meaning their color wouldn't change after creation. If immutability is desired, remove this setter.